src/hoc18/Program2.java

```diff
@@ -0,0 +1,26 @@
+package hoc18;
+
+public class Program2 {
```

Comment:
```suggestion
public class DecodeString {
```
Class name should indicate the function of the class. DecodeString would have been a better name in this case.